Create process.rst (#12)
dimagi/open-source: docs/process.rst

Code Release Process
^^^^^^^^^^^^^^^^^^^^

> This document contains guidance on internal processes that can be relevant when submitting code, in particular to the `commcare-hq <https://github.com/dimagi/commcare-hq/>`_ repository.

Reviewer: ```suggestion
This document contains guidance on internal processes that are commonly part of code submission and review in our Open Source repositories.
```

PR Template
^^^^^^^^^^^

> Depending on the repository, your PR may be populated with a structured template. Most of this is oriented around clearly communicating the change being made and accounting for risks in the change.

Reviewer: ```suggestion
Depending on the repository, your PR may be populated with a structured template, or in some cases you may have a choice of templates to apply. 

We use templates for a mix of applying guiderails to the submission process, as well as ensuring that the structure of changes is clearly communicated including risks and how they are accounted for.
```

Labels
^^^^^^

> The main Dimagi repos require PRs to be labeled for the sake of internal process. This is enforced by the `label-bot <https://github.com/dimagi/label-bot>`_, which will mark your PR as failing a check if it is unlabeled. A member with write access will label your PR.

Reviewer: ```suggestion
Many of our repos require PRs to be labeled for the sake of internal process, with the labels themselves being specific to the contribution process for each repository. 

Label adherence is enforced by the `label-bot <https://github.com/dimagi/label-bot>`_, which will mark your PR as failing a check if it is unlabeled. 

If you do not have core committing access, a member with write access will label your PR based on the code changed, your PR overview, and any other factors that come up during review.
```
